web/server/api/endpoints/files.py

```diff
@@ -67,18 +67,22 @@ async def write_file(
         replace_file(settings.project_path / path, settings.project_path / path_or_new_path)
     else:
         full_path = settings.project_path / path
-        format_file_status = models.FormatFileStatus(
-            status=models.Status.INIT, path=path_or_new_path
-        )
-        if content and Path(path_or_new_path).suffix == ".sql":
+        config = context.config_for_path(Path(path_or_new_path))
+        if config.ui.format_on_save and content and Path(path_or_new_path).suffix == ".sql":
+            format_file_status = models.FormatFileStatus(
+                status=models.Status.INIT, path=path_or_new_path
+            )
             path_to_model_mapping = await get_path_to_model_mapping(settings=settings)
             model = path_to_model_mapping.get(Path(full_path))
-            default_dialect = context.config_for_path(Path(path_or_new_path)).dialect
+            default_dialect = config.dialect
             dialect = model.dialect if model and model.is_sql else default_dialect
-
             try:
                 expressions = parse(content, default_dialect=default_dialect)
-                content = format_model_expressions(expressions, dialect)
+                content = format_model_expressions(
+                    expressions, dialect, **config.ui.format_options.dict(exclude={"newline"})
+                )
+                if config.ui.format_options.newline:
```

Comment:
why do we need this?

Reply:
To match the CLI and because by default the formatter strips trailing newlines completely, or rather doesn't generate them. And having a single trailing newline is a style preference some users (such as myself) may have. Also it makes tools like `cat`, `wc`, and likely others work properly.

Reply:
We could name it append_newline everywhere if preferable, to make its intention clearer @izeigerman